app/routes/datasets/versions.py

```diff
@@ -240,13 +240,36 @@ async def append_to_version(
     # For the background task, we only need the new source uri from the request
     input_data = {"creation_options": deepcopy(default_asset.creation_options)}
     input_data["creation_options"]["source_uri"] = request.source_uri
+
+    # If source_driver is "text", this is a datapump request
+    if input_data["creation_options"]["source_driver"] != "text":
+        # Verify that source_driver is not None
+        if input_data["creation_options"]["source_driver"] is None:
+            raise HTTPException(
+                status_code=400,
+                detail="Source driver must be specified for non-datapump requests."
+            )
+
+        # Append the new layers to the existing ones
+        if input_data["creation_options"].get("layers") is None: # ERROR: layers is not defined
+            input_data["creation_options"]["layers"] = request.layers
+        elif request.layers is not None:
+            input_data["creation_options"]["layers"] += request.layers
+        else:
+            input_data["creation_options"]["layers"] = request.layers
+            
     background_tasks.add_task(
         append_default_asset, dataset, version, input_data, default_asset.asset_id
     )
 
     # We now want to append the new uris to the existing ones and update the asset
     update_data = {"creation_options": deepcopy(default_asset.creation_options)}
-    update_data["creation_options"]["source_uri"] += request.source_uri
+    update_data["creation_options"]["source_uri"] += request.source_uri # ERROR: only one source_uri is allowed
+    if input_data["creation_options"].get("layers") is not None:
+        if update_data["creation_options"]["layers"] is not None:
+            update_data["creation_options"]["layers"] += request.layers
+        else:
+            update_data["creation_options"]["layers"] = request.layers
```

Comment:
It's confusing that this logic here is different from the logic above for input_data[].  Can you possibly add in a comment on what you expect the difference to be for input_data["creation_options"]["layers" and update_data["creation_options"]["layers"]?

Reply:
I added comments to reflect this. `input_data["creation_options"]` is used for the append operation and `update_data["creation_options"]` is used to update the version's creation options to reflect changes made in the append request.

Reply:
Great, thanks for the added comments to clarify!  The logic in both cases looks better now.